### PR TITLE
Use kubernetes informer instead of watcher

### PR DESCRIPTION
**Pull request by victor-carvalho**
Using informers gives us access to many tools in client-go and also takes care of many things that we need to do like worrying about reconnections.

### CODE REVIEW

File: controller/utils.go [hunk 1]
```diff
@@ -255,6 +255,8 @@ func replacePodWorker(
 			}
 
 			logger.Infof("pods remaining to replace: %d", len(pods))
+
+			return nil
```
Comment (leohahn) [inline controller/utils.go:259]: is the for loop necessary if you're returning nil here?

Comment (victor-carvalho) [reply controller/utils.go:259]: Sorry, forgot the if.

File: watcher/watcher.go [hunk 2]
```diff
@@ -1513,94 +1520,128 @@ func (w *Watcher) checkIfUsageIsAboveLimit(
...
+	if !ok {
+		// Object is not a string so we skip, shouldn't happen
+		w.Queue.Forget(obj)
+		return
+	}
+
+	err := w.podEventHandler(key)
+	if err != nil {
+		// An error happened so we requeue it so it can processed again later
+		w.Queue.AddRateLimited(key)
```
Comment (leohahn) [inline watcher/watcher.go:1595]: Do we want to reprocess in all error cases?

Comment (victor-carvalho) [reply watcher/watcher.go:1595]: Even though I treat some different errors in handler code the only errors that can really happen are redis errors and I think we always want to retry in this cases or redis state can become too different from Kubernetes and that's really bad for maestro.

**Comment by leohahn** [comment]
Does the informer take care of reconnections, etc?

File: watcher/watcher.go (hunk 2 above)
Comment (leohahn) [reply watcher/watcher.go:1595]: Cool, makes sense.

**Comment by victor-carvalho** [comment]
> Does the informer take care of reconnections, etc?

Yes.

**Review by gabrielcorado — APPROVED** [review]

**Pull request merged**